x/evm/keeper/statedb.go

```diff
@@ -608,10 +608,10 @@ func (k *Keeper) AddLog(log *ethtypes.Log) {
 	log.Index = uint(k.GetLogSizeTransient())
 	k.IncreaseLogSizeTransient()
 
-	logs := k.GetTxLogs(log.TxHash)
-	logs = append(logs, log)
-
-	k.SetLogs(log.TxHash, logs)
+	s := prefix.NewStore(k.Ctx().KVStore(k.storeKey), types.KeyPrefixTxLogNum)
+	bi := new(big.Int).SetBytes(s.Get(log.TxHash.Bytes()))
+	s.Set(log.TxHash.Bytes(), new(big.Int).SetInt64(bi.Int64()+1).Bytes())
```

Comment:
curious on why are you using the big.Int to set the hash?

Reply:
just bytes would be simpler

Reply:
> curious on why are you using the big.Int to set the hash?

Mainly want to bank the space and 2 liners,  so most of the cases will store only 1 byte, I think fewer cases might need to use 2 bytes(more than 255 logs in a transaction)

I can add a helper to deal with the dynamic bytes transfer, or do you think we can always use 2 bytes to store the transaction log count?